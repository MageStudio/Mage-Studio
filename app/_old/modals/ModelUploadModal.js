import React, { Component } from 'react';
import { connect } from 'react-redux';
importÂ { Modal } from 'antd';

import {
    uploadModel,
    getModels,
    hideModelUploadModal,
    loadSingleModel
} from '../actions/models';

import Footer from './footer';
import ModelsSelector from './content/ModelsSelector';
import FileUploaderBox from './content/FileUploaderBox';

import './modals.scss';
import '../../lib/style.scss';

export class ModelUploadModal extends Component {

    constructor(props) {
        super(props);

        this.state = {
            file: false,
            selection: false
        };
    }

    componentDidUpdate(prevProps) {
        if (prevProps.visible !== this.props.visible && this.props.visible) {
            this.fetchModels();
            this.setState({ file: false, selection: false });
        }
    }

    componentDidMount() {
        this.fetchModels();
    }

    fetchModels = () => {
        const { getModels = f => f, config } = this.props;
        const { project } = config;

        getModels(project);
    };

    getFooter = (loading) => (
        <Footer
            loading={loading}
            onCancel={this.handleCancel}
            onConfirm={this.handleConfirm}
            canCancel
        />
    );

    handleCancel = () => {
        const { onDismiss } = this.props;

        onDismiss();
    };

    handleConfirm = () => {
        const { loadSingleModel, config, uploadModel } = this.props;
        const { selection, file } = this.state;
        const { project } = config;

        if (file) {
            uploadModel(project, file);
        } else if (selection) {
            loadSingleModel(project, selection);
        }
    }

    handleOnSelect = (selection) => {
        this.setState({ selection });
    }

    handleBeforeUpload = (file) => {
        this.setState({ file });

        return false;
    }

    getContent = () => {
        const { list } = this.props;
        const { selection } = this.state;

        return (
            <div className='box row'>
                <ModelsSelector
                    onSelect={this.handleOnSelect}
                    selection={selection}
                    list={list}
                />
                <FileUploaderBox
                    className={'models'}
                    onBeforeUpload={this.handleBeforeUpload}
                    name={"data"} />
            </div>
        );
    }

    render() {
        const { visible, loading } = this.props;

        return (
            <Modal
                className='modal big'
                title="Models"
                visible={visible}
                onCancel={this.handleCancel}
                footer={this.getFooter(loading)}>
                { this.getContent() }
            </Modal>
        )
    }
}

const mapStateToProps = (state = {}) => {
    const { models, config } = state;
    const {
        visible,
        loading = false,
        error = false,
        completed = false,
        data = false,
        list = []
    } = models;

    return {
        config,
        visible,
        loading,
        completed,
        data,
        list,
        error,
    }
};

const mapDispatchToProps = (dispatch) => ({
    getModels: (project) => dispatch(getModels(project)),
    uploadModel: (project, payload) => dispatch(uploadModel(project, payload)),
    onDismiss: () => dispatch(hideModelUploadModal()),
    loadSingleModel: (project, name) => dispatch(loadSingleModel(project, name))
});

export default connect(mapStateToProps, mapDispatchToProps)(ModelUploadModal);
