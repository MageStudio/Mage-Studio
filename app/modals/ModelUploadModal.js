import React from 'react';
import { connect } from 'react-redux';
import { Modal, Input, Button } from 'antd';

import {
    uploadModel,
    getModels,
    hideModelUploadModal,
    loadSingleModel
} from '../actions/modelModal';

import Footer from './footer';
import Divider from './content/Divider';
import ModelsSelector from './content/ModelsSelector';
import ModelUploader from './content/ModelUploader';

import { PROJECTS_URL } from '../lib/constants';

import './modals.scss';
import '../style.scss';

class ModelUploadModal extends React.Component {

    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const { getModels = f => f, config, visible } = this.props;
        const { project } = config;

        if (visible) {
            getModels(project);
        }
    }

    getFooter = (loading) => (
        <Footer
            loading={loading}
            onCancel={this.handleCancel}
            onConfirm={this.handleConfirm}
            canCancel
        />
    )

    handleCancel = () => {
        const { hideModelUploadModal } = this.props;

        hideModelUploadModal();
    }

    handleConfirm = () => {}

    handleOnSelect = (name) => () => {
        const { loadSingleModel, config } = this.props;
        const { project } = config;

        loadSingleModel(project, name);
    }

    handleBeforeUpload = (file) => {
        const { uploadModel, config } = this.props;
        const { project } = config;

        uploadModel(project, file);

        return false;
    }

    getContent = () => {
        const { list, data } = this.props;
        return (
            <div className='box row'>
                <ModelsSelector
                    onSelect={this.handleOnSelect}
                    uploaded={data}
                    list={list}
                />
                <ModelUploader
                    onBeforeUpload={this.handleBeforeUpload}
                    name={"data"} />
            </div>
        );
    }

    render() {
        const { visible, loading } = this.props;

        return (
            <Modal
                className='modal big'
                title="Models"
                visible={visible}
                onCancel={this.handleCancel}
                footer={this.getFooter(loading)}>
                { this.getContent() }
            </Modal>
        )
    }
}

const mapStateToProps = (state = {}) => {
    const { modelModal, config } = state;
    const {
        visible,
        loading = false,
        error = false,
        completed = false,
        data = false,
        list = []
    } = modelModal;

    return {
        config,
        visible,
        loading,
        completed,
        data,
        list,
        error,
    }
};

const mapDispatchToProps = (dispatch) => ({
    getModels: (project) => dispatch(getModels(project)),
    uploadModel: (project, payload) => dispatch(uploadModel(project, payload)),
    hideModelUploadModal: () => dispatch(hideModelUploadModal()),
    loadSingleModel: (project, name) => dispatch(loadSingleModel(project, name))
});

export default connect(mapStateToProps, mapDispatchToProps)(ModelUploadModal);
