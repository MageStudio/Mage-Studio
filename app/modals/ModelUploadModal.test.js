import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { ModelUploadModal } from './ModelUploadModal';

const defaultProps = {
    config: {}
};

const getModal = (props) => shallow(<ModelUploadModal {...defaultProps} {...props} />);

describe('ModelUploadModal', () => {

    it('should render fine', () => {
        expect(getModal()).toMatchSnapshot();
    });

    it('should have right initial state', () => {
        const component = getModal();

        expect(component.state().file).toBe(false);
        expect(component.state().selection).toBe(false);
    });

    it('should call getModels on componentDidMount', () => {
        const spy = sinon.spy();
        const component = getModal({ getModels: spy });

        expect(spy.called).toBe(true);
    });

    describe('events', () => {

        it('should set the right state after selection', () => {
            const component = getModal();
            const selector = component.find('ModelsSelector');

            selector.simulate('select', 'selection');

            expect(component.state().selection).toBe('selection');
        });

        it('should set the right state after file is about to upload', () => {
            const component = getModal();
            const selector = component.find('FileUploaderBox');

            selector.simulate('beforeUpload', 'filename.js');

            expect(component.state().file).toBe('filename.js');
        });

        it('should call onDismiss when the modal is cancelled', () => {
            const onDismiss = sinon.spy();
            const component = getModal({ onDismiss });
            const modal = component.find('Modal');

            modal.simulate('cancel');

            expect(onDismiss.called).toBe(true);
        });

        it('should call uploadModel when user confirms and hes trying to upload', () => {
            const uploadModel = sinon.spy();
            const config = { project: 'test' };
            const component = getModal({ uploadModel, config });

            component.setState({ file: 'filename.js' });

            component.instance().handleConfirm();

            expect(uploadModel.calledWith('test', 'filename.js')).toBe(true);
        });

        it('should call loadSingleModel when user confirms selection', () => {
            const loadSingleModel = sinon.spy();
            const config = { project: 'test' };
            const component = getModal({ loadSingleModel, config });

            component.setState({ selection: 'filename.js' });

            component.instance().handleConfirm();

            expect(loadSingleModel.calledWith('test', 'filename.js')).toBe(true);
        });
    });
});
