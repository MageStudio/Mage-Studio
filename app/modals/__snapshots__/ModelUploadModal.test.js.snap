// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelUploadModal should render fine 1`] = `
<Modal
  className="modal big"
  confirmLoading={false}
  footer={
    <Footer
      canCancel={true}
      onCancel={[Function]}
      onConfirm={[Function]}
    />
  }
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Models"
  transitionName="zoom"
  visible={false}
  width={520}
>
  <div
    className="box row"
  >
    <ModelsSelector
      onSelect={[Function]}
      selection={false}
    />
    <FileUploaderBox
      className="models"
      name="data"
      onBeforeUpload={[Function]}
    />
  </div>
</Modal>
`;
